### PR TITLE
Fix/1896/file modifications

### PR DESCRIPTION
**Pull request by jmshark**
## Proposed changes
Fix for [1896](https://github.com/thin-edge/thin-edge.io/issues/1896) sed creates a temp file where the edits are made then moves the file to the final destination. Moving a file in the same directory generates various [EventKind::Modify](https://docs.rs/notify/5.1.0/notify/event/enum.EventKind.html) where the modification kind is Name these events were being dropped by `crates\common\tedge_utils\src\notify.rs` which is why it fails for these cases.

In addition to capturing these events I've also removed some dead code.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by reubenmiller** [comment]
@jmshark We'll have to rebase once https://github.com/thin-edge/thin-edge.io/pull/1875 is merged (it is going through review now), as it includes a test-stability fix. Though you should still check that the configuration tests marked as skipped have been removed (during rebasing/resolving conflicts).

**Comment by jmshark** [comment]
> > @jmshark We'll have to rebase once #1875 is merged (it is going through review now), as it includes a test-stability fix. Though you should still check that the configuration tests marked as skipped have been removed (during rebasing/resolving conflicts).
> 
> @jmshark #1875 has been merged now, so please rebase/resolve conflicts.

Done

File: tests/RobotFramework/tests/cumulocity/configuration/configuration_operation.robot [hunk 1]
```diff
@@ -72,7 +72,6 @@ Update configuration plugin config via cloud
 Modify configuration plugin config via local filesystem modify inplace
     [Documentation]    Editing the c8y-configuration-plugin.toml locally requires the
     ...                c8y-configuration-plugin service to be restarted
```
Comment (reubenmiller) [inline tests/RobotFramework/tests/cumulocity/configuration/configuration_operation.robot:74]: We can remove the documentation comment now (as it is no longer applicable)
```suggestion
```

File: crates/common/tedge_utils/src/notify.rs [hunk 2]
```diff
@@ -70,142 +61,77 @@ impl NotifyStream {
...
     }
 
-    pub fn add_watcher(
-        &mut self,
-        dir_path: &Path,
-        file: Option<String>,
-        events: &[FsEvent],
-    ) -> Result<(), NotifyStreamError> {
+    /// Will return an error if you try to watch a file/directory which doesn't exist
+    pub fn add_watcher(&mut self, dir_path: &Path) -> Result<(), NotifyStreamError> {
```
Comment (didier-wenzek) [inline crates/common/tedge_utils/src/notify.rs:111]: Removing the list of watch events is a nice simplification. The caller might receive more messages than expected but it far simpler to do the filtering there.

**Review by didier-wenzek — APPROVED** [review]
Approved. Nice fix and simplification.

**Pull request merged**